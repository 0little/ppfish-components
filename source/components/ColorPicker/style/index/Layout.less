@prefixClassName: ~'@{prefixClass}-panel';

.@{prefixClassName}-inner {
  position: relative;
  border-radius: @border-radius-lg;
  box-shadow: @box-shadow-base;
  border: 1px solid #ebedf0;
  padding-bottom: 8px;
  padding-top: 3px;
}

.@{prefixClassName}-wrap {
  margin: 5px 0 0;
  height: 26px;
  width: 100%;
  position: relative;
  &-preview {
    position: absolute;
    right: 8px;
  }
  &-ribbon {
    position: absolute;
    left: 8px;
    top: 0;
    right: 41px;
    height: 26px;
  }
  &-alpha {
    position: absolute;
    left: 8px;
    right: 41px;
    bottom: 0;
    height: (26px - 5px) / 2;
  }
}

.@{prefixClassName}-wrap-has-alpha {
  .@{prefixClassName}-wrap-ribbon {
    height: (26px - 5px) / 2;
  }
}
