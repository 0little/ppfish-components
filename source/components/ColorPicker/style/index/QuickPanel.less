.@{prefixClass}-quick-panel {
  width: 200px;
  background-color: #fff;
  box-sizing: border-box;
  outline: none;
  z-index: 9;
  border-radius: 4px;
  -moz-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 8px;
  text-align: center;
  border: @border-width-base @border-style-base @border-color-base;
  * {
    box-sizing: border-box;
  }
  &-open {
    display: block;
  }
  &-close {
    display: none;
  }

  &-color-spn {
    float: left;
    width: 30px;
    height: 30px;
    margin: 3px;
    cursor: pointer;
  }

  &-color-spn-active {
    .iconfont-mixin();
    vertical-align: top;
    &:before {
      content: "\e671";
      line-height: 30px;
      color: #fff;
    }
  }
  &-inner {
    .clearfix();
  }
  &-custom-btn {
    margin: 3px;
    cursor: pointer;
    float: left;
    text-align: center;
    background: #fff;
    font-size: 12px;
    .clearfix();
    &-color {
      width: 30px;
      height: 30px;
      float: left;
      .iconfont-mixin();
      vertical-align: top;
      &:before {
        line-height: 30px;
        color: #fff;
        content: "\e671";
      }
    }
    &-text {
      height: 30px;
      border: @border-width-base @border-style-base @border-color-base;
      float: left;
      display: inline-block;
      width: 60px;
      line-height: 30px;
      text-align: center;
    }
  }
}
