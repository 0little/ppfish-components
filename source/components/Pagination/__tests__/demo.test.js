import { getPlainComponentList } from "../../../../site/utils/index.js";
const plainComponentList = getPlainComponentList();

let demo_component = 'pagination';

let demoName = demo_component.substr(0, 1).toLowerCase() + demo_component.substr(1);
const menuItem = plainComponentList.find(item => item.key === demoName);

if(menuItem){
  const demoTest = require('../../../tests/demoTest').default;
  demoTest(demoName);
}else{
  xit('skip',()=>{});
}
