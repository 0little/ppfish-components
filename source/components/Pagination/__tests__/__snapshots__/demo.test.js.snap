// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders pagination demo correctly 1`] = `
<div>
  <ul
    class="fishd-pagination "
    unselectable="unselectable"
  >
    <li
      aria-disabled="true"
      class="fishd-pagination-disabled fishd-pagination-prev"
      title="上一页"
    >
      <a
        class="fishd-pagination-item-link"
      />
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-1 fishd-pagination-item-active"
      tabindex="0"
      title="1"
    >
      <a>
        1
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-2"
      tabindex="0"
      title="2"
    >
      <a>
        2
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-3"
      tabindex="0"
      title="3"
    >
      <a>
        3
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-4"
      tabindex="0"
      title="4"
    >
      <a>
        4
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-5"
      tabindex="0"
      title="5"
    >
      <a>
        5
      </a>
    </li>
    <li
      aria-disabled="false"
      class=" fishd-pagination-next"
      tabindex="0"
      title="下一页"
    >
      <a
        class="fishd-pagination-item-link"
      />
    </li>
  </ul>
  <ul
    class="fishd-pagination "
    unselectable="unselectable"
  >
    <li
      aria-disabled="false"
      class=" fishd-pagination-prev"
      tabindex="0"
      title="上一页"
    >
      <a
        class="fishd-pagination-item-link"
      />
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-1"
      tabindex="0"
      title="1"
    >
      <a>
        1
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-2"
      tabindex="0"
      title="2"
    >
      <a>
        2
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-3 fishd-pagination-item-active"
      tabindex="0"
      title="3"
    >
      <a>
        3
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-4"
      tabindex="0"
      title="4"
    >
      <a>
        4
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-5 fishd-pagination-item-before-jump-next"
      tabindex="0"
      title="5"
    >
      <a>
        5
      </a>
    </li>
    <li
      class="fishd-pagination-jump-next"
      tabindex="0"
      title="向后 5 页"
    >
      <a
        class="fishd-pagination-item-link"
      />
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-50"
      tabindex="0"
      title="50"
    >
      <a>
        50
      </a>
    </li>
    <li
      aria-disabled="false"
      class=" fishd-pagination-next"
      tabindex="0"
      title="下一页"
    >
      <a
        class="fishd-pagination-item-link"
      />
    </li>
    <li
      class="fishd-pagination-options"
    >
      <div
        class="fishd-select fishd-pagination-options-size-changer"
      >
        <div
          class="fishd-select-content"
        >
          <div
            class="fishd-select-caret"
          >
            <i
              class="fishdicon fishdicon-down-fill"
            />
          </div>
          <span
            class="fishd-select-option-single"
          >
            10 条/页
          </span>
        </div>
      </div>
    </li>
  </ul>
  <ul
    class="fishd-pagination "
    unselectable="unselectable"
  >
    <li
      aria-disabled="false"
      class=" fishd-pagination-prev"
      tabindex="0"
      title="上一页"
    >
      <a
        class="fishd-pagination-item-link"
      />
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-1"
      tabindex="0"
      title="1"
    >
      <a>
        1
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-2"
      tabindex="0"
      title="2"
    >
      <a>
        2
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-3 fishd-pagination-item-active"
      tabindex="0"
      title="3"
    >
      <a>
        3
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-4"
      tabindex="0"
      title="4"
    >
      <a>
        4
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-5"
      tabindex="0"
      title="5"
    >
      <a>
        5
      </a>
    </li>
    <li
      aria-disabled="false"
      class=" fishd-pagination-next"
      tabindex="0"
      title="下一页"
    >
      <a
        class="fishd-pagination-item-link"
      />
    </li>
  </ul>
  <ul
    class="fishd-pagination "
    unselectable="unselectable"
  >
    <li
      aria-disabled="true"
      class="fishd-pagination-disabled fishd-pagination-prev"
      title="上一页"
    >
      <a>
        上一页
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-1 fishd-pagination-item-active"
      tabindex="0"
      title="1"
    >
      <a>
        1
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-2"
      tabindex="0"
      title="2"
    >
      <a>
        2
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-3"
      tabindex="0"
      title="3"
    >
      <a>
        3
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-4"
      tabindex="0"
      title="4"
    >
      <a>
        4
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-5 fishd-pagination-item-before-jump-next"
      tabindex="0"
      title="5"
    >
      <a>
        5
      </a>
    </li>
    <li
      class="fishd-pagination-jump-next"
      tabindex="0"
      title="向后 5 页"
    >
      <a
        class="fishd-pagination-item-link"
      />
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-50"
      tabindex="0"
      title="50"
    >
      <a>
        50
      </a>
    </li>
    <li
      aria-disabled="false"
      class=" fishd-pagination-next"
      tabindex="0"
      title="下一页"
    >
      <a>
        下一页
      </a>
    </li>
  </ul>
  <ul
    class="fishd-pagination "
    unselectable="unselectable"
  >
    <li
      aria-disabled="false"
      class=" fishd-pagination-prev"
      tabindex="0"
      title="上一页"
    >
      <a
        class="fishd-pagination-item-link"
      />
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-1"
      tabindex="0"
      title="1"
    >
      <a>
        1
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-2 fishd-pagination-item-active"
      tabindex="0"
      title="2"
    >
      <a>
        2
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-3"
      tabindex="0"
      title="3"
    >
      <a>
        3
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-4"
      tabindex="0"
      title="4"
    >
      <a>
        4
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-5 fishd-pagination-item-before-jump-next"
      tabindex="0"
      title="5"
    >
      <a>
        5
      </a>
    </li>
    <li
      class="fishd-pagination-jump-next"
      tabindex="0"
      title="向后 5 页"
    >
      <a
        class="fishd-pagination-item-link"
      />
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-50"
      tabindex="0"
      title="50"
    >
      <a>
        50
      </a>
    </li>
    <li
      aria-disabled="false"
      class=" fishd-pagination-next"
      tabindex="0"
      title="下一页"
    >
      <a
        class="fishd-pagination-item-link"
      />
    </li>
    <li
      class="fishd-pagination-options"
    >
      <div
        class="fishd-pagination-options-quick-jumper"
      >
        跳至
        <input
          type="text"
          value=""
        />
        页
      </div>
    </li>
  </ul>
  <div
    id="components-pagination-demo-mini"
  >
    <ul
      class="fishd-pagination mini"
      style="margin-bottom:24px"
      unselectable="unselectable"
    >
      <li
        aria-disabled="true"
        class="fishd-pagination-disabled fishd-pagination-prev"
        title="上一页"
      >
        <a
          class="fishd-pagination-item-link"
        />
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-1 fishd-pagination-item-active"
        tabindex="0"
        title="1"
      >
        <a>
          1
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-2"
        tabindex="0"
        title="2"
      >
        <a>
          2
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-3"
        tabindex="0"
        title="3"
      >
        <a>
          3
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-4"
        tabindex="0"
        title="4"
      >
        <a>
          4
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-5"
        tabindex="0"
        title="5"
      >
        <a>
          5
        </a>
      </li>
      <li
        aria-disabled="false"
        class=" fishd-pagination-next"
        tabindex="0"
        title="下一页"
      >
        <a
          class="fishd-pagination-item-link"
        />
      </li>
    </ul>
    <ul
      class="fishd-pagination mini"
      style="margin-bottom:24px"
      unselectable="unselectable"
    >
      <li
        aria-disabled="true"
        class="fishd-pagination-disabled fishd-pagination-prev"
        title="上一页"
      >
        <a
          class="fishd-pagination-item-link"
        />
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-1 fishd-pagination-item-active"
        tabindex="0"
        title="1"
      >
        <a>
          1
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-2"
        tabindex="0"
        title="2"
      >
        <a>
          2
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-3"
        tabindex="0"
        title="3"
      >
        <a>
          3
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-4"
        tabindex="0"
        title="4"
      >
        <a>
          4
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-5"
        tabindex="0"
        title="5"
      >
        <a>
          5
        </a>
      </li>
      <li
        aria-disabled="false"
        class=" fishd-pagination-next"
        tabindex="0"
        title="下一页"
      >
        <a
          class="fishd-pagination-item-link"
        />
      </li>
      <li
        class="fishd-pagination-options"
      >
        <div
          class="fishd-select fishd-pagination-options-size-changer fishd-select-small"
        >
          <div
            class="fishd-select-content"
          >
            <div
              class="fishd-select-caret"
            >
              <i
                class="fishdicon fishdicon-down-fill"
              />
            </div>
            <span
              class="fishd-select-option-single"
            >
              10 条/页
            </span>
          </div>
        </div>
        <div
          class="fishd-pagination-options-quick-jumper"
        >
          跳至
          <input
            type="text"
            value=""
          />
          页
        </div>
      </li>
    </ul>
    <ul
      class="fishd-pagination mini"
      unselectable="unselectable"
    >
      <li
        class="fishd-pagination-total-text"
      >
        共 50 条数据
      </li>
      <li
        aria-disabled="true"
        class="fishd-pagination-disabled fishd-pagination-prev"
        title="上一页"
      >
        <a
          class="fishd-pagination-item-link"
        />
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-1 fishd-pagination-item-active"
        tabindex="0"
        title="1"
      >
        <a>
          1
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-2"
        tabindex="0"
        title="2"
      >
        <a>
          2
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-3"
        tabindex="0"
        title="3"
      >
        <a>
          3
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-4"
        tabindex="0"
        title="4"
      >
        <a>
          4
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-5"
        tabindex="0"
        title="5"
      >
        <a>
          5
        </a>
      </li>
      <li
        aria-disabled="false"
        class=" fishd-pagination-next"
        tabindex="0"
        title="下一页"
      >
        <a
          class="fishd-pagination-item-link"
        />
      </li>
    </ul>
  </div>
  <ul
    class="fishd-pagination "
    unselectable="unselectable"
  >
    <li
      aria-disabled="false"
      class=" fishd-pagination-prev"
      tabindex="0"
      title="上一页"
    >
      <a
        class="fishd-pagination-item-link"
      />
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-1"
      tabindex="0"
      title="1"
    >
      <a>
        1
      </a>
    </li>
    <li
      class="fishd-pagination-jump-prev"
      tabindex="0"
      title="向前 5 页"
    >
      <a
        class="fishd-pagination-item-link"
      />
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-4 fishd-pagination-item-after-jump-prev"
      tabindex="0"
      title="4"
    >
      <a>
        4
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-5"
      tabindex="0"
      title="5"
    >
      <a>
        5
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-6 fishd-pagination-item-active"
      tabindex="0"
      title="6"
    >
      <a>
        6
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-7"
      tabindex="0"
      title="7"
    >
      <a>
        7
      </a>
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-8 fishd-pagination-item-before-jump-next"
      tabindex="0"
      title="8"
    >
      <a>
        8
      </a>
    </li>
    <li
      class="fishd-pagination-jump-next"
      tabindex="0"
      title="向后 5 页"
    >
      <a
        class="fishd-pagination-item-link"
      />
    </li>
    <li
      class="fishd-pagination-item fishd-pagination-item-50"
      tabindex="0"
      title="50"
    >
      <a>
        50
      </a>
    </li>
    <li
      aria-disabled="false"
      class=" fishd-pagination-next"
      tabindex="0"
      title="下一页"
    >
      <a
        class="fishd-pagination-item-link"
      />
    </li>
  </ul>
  <ul
    class="fishd-pagination fishd-pagination-simple "
  >
    <li
      aria-disabled="false"
      class=" fishd-pagination-prev"
      tabindex="0"
      title="上一页"
    >
      <a
        class="fishd-pagination-item-link"
      />
    </li>
    <li
      class="fishd-pagination-simple-pager"
      title="2/5"
    >
      <input
        size="3"
        type="text"
        value="2"
      />
      <span
        class="fishd-pagination-slash"
      >
        ／
      </span>
      5
    </li>
    <li
      aria-disabled="false"
      class=" fishd-pagination-next"
      tabindex="0"
      title="下一页"
    >
      <a
        class="fishd-pagination-item-link"
      />
    </li>
  </ul>
  <div>
    <ul
      class="fishd-pagination "
      unselectable="unselectable"
    >
      <li
        class="fishd-pagination-total-text"
      >
        共 85 条数据
      </li>
      <li
        aria-disabled="true"
        class="fishd-pagination-disabled fishd-pagination-prev"
        title="上一页"
      >
        <a
          class="fishd-pagination-item-link"
        />
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-1 fishd-pagination-item-active"
        tabindex="0"
        title="1"
      >
        <a>
          1
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-2"
        tabindex="0"
        title="2"
      >
        <a>
          2
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-3"
        tabindex="0"
        title="3"
      >
        <a>
          3
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-4"
        tabindex="0"
        title="4"
      >
        <a>
          4
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-5"
        tabindex="0"
        title="5"
      >
        <a>
          5
        </a>
      </li>
      <li
        aria-disabled="false"
        class=" fishd-pagination-next"
        tabindex="0"
        title="下一页"
      >
        <a
          class="fishd-pagination-item-link"
        />
      </li>
    </ul>
    <br />
    <ul
      class="fishd-pagination "
      unselectable="unselectable"
    >
      <li
        class="fishd-pagination-total-text"
      >
        第1-20 条，共 85 条数据
      </li>
      <li
        aria-disabled="true"
        class="fishd-pagination-disabled fishd-pagination-prev"
        title="上一页"
      >
        <a
          class="fishd-pagination-item-link"
        />
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-1 fishd-pagination-item-active"
        tabindex="0"
        title="1"
      >
        <a>
          1
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-2"
        tabindex="0"
        title="2"
      >
        <a>
          2
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-3"
        tabindex="0"
        title="3"
      >
        <a>
          3
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-4"
        tabindex="0"
        title="4"
      >
        <a>
          4
        </a>
      </li>
      <li
        class="fishd-pagination-item fishd-pagination-item-5"
        tabindex="0"
        title="5"
      >
        <a>
          5
        </a>
      </li>
      <li
        aria-disabled="false"
        class=" fishd-pagination-next"
        tabindex="0"
        title="下一页"
      >
        <a
          class="fishd-pagination-item-link"
        />
      </li>
    </ul>
  </div>
</div>
`;
